timemachine/cpp/src/kernels/k_flat_bottom_bond.cuh

```diff
@@ -1,6 +1,65 @@
 #include "../fixed_point.hpp"
 #include "k_fixed_point.cuh"
 
+// branchless implementation of piecewise function
+template <typename RealType>
+RealType __device__ __forceinline__ compute_flat_bottom_energy(
+    RealType k, RealType r_lt_rmin, RealType r_gt_rmax, RealType r, RealType rmin, RealType rmax) {
+    return (k / 4) * (r_lt_rmin * (pow(r - rmin, 4)) + r_gt_rmax * (pow(r - rmax, 4)));
+}
+
+template <typename RealType>
+void __global__ k_log_probability_selection(
+    const int N,                             // Num atoms
+    const double kBt,                        // BOLTZ * temperature
+    const float radius,                      // Radius, corresponds to r_max for flat bottom
+    const float k,                           // Constant restraint value
+    const unsigned int reference_idx,        // Idx that the probability is specific to
+    const double *__restrict__ coords,       // [N, 3]
+    const double *__restrict__ box,          // [3, 3]
+    const float *__restrict__ probabilities, // [N] probabilities of selection
+    unsigned int *__restrict__ selected      // [N] idx array, N if idx is not selected, else idx of coordinate
+) {
+    const unsigned int idx = blockDim.x * blockIdx.x + threadIdx.x;
+    if (idx >= N) {
+        return;
+    }
+    const float radius_sq = radius * radius;
+    const RealType bx = box[0 * 3 + 0];
+    const RealType by = box[1 * 3 + 1];
+    const RealType bz = box[2 * 3 + 2];
```

Comment:
Should we throw an exception in `FlatBottomBond::execute_device` if passed a non-ortholinear box, as we do for the nonbonded potentials?

Reply:
Seems like this would be well suited to https://github.com/proteneer/timemachine/pull/916

We should probably verify non-ortholinear boxes as we go from python -> C++ and avoid runtime checks.

Reply:
Makes sense to add to #916, assuming that all potentials require `box` to be diagonal. Just noting that there's currently an inconsistency with the nonbonded potentials, which check this assumption in `execute_device`.

Reply:
@mcwitt Was resolved in #916